commit 3959df1dfb9538498ec3372a2d390bc7fbdbfac2
Merge: 41d859a83c56 6c36dfe94918
Author: Linus Torvalds <torvalds@linux-foundation.org>
Date:   Mon Aug 31 20:20:30 2015 -0700

    Merge branch 'ras-core-for-linus' of git://git.kernel.org/pub/scm/linux/kernel/git/tip/tip
    
    Pull RAS updates from Ingo Molnar:
     "MCE handling updates, but also some generic drivers/edac/ changes to
      better organize the Kconfig space"
    
    * 'ras-core-for-linus' of git://git.kernel.org/pub/scm/linux/kernel/git/tip/tip:
      x86/ras: Move AMD MCE injector to arch/x86/ras/
      x86/mce: Add a wrapper around mce_log() for injection
      x86/mce: Rename rcu_dereference_check_mce() to mce_log_get_idx_check()
      RAS: Add a menuconfig option with descriptive text
      x86/mce: Reenable CMCI banks when swiching back to interrupt mode
      x86/mce: Clear Local MCE opt-in before kexec
      x86/mce: Remove unused function declarations
      x86/mce: Kill drain_mcelog_buffer()
      x86/mce: Avoid potential deadlock due to printk() in MCE context
      x86/mce: Remove the MCE ring for Action Optional errors
      x86/mce: Don't use percpu workqueues
      x86/mce: Provide a lockless memory pool to save error records
      x86/mce: Reuse one of the u16 padding fields in 'struct mce'

diff --cc arch/x86/kernel/cpu/mcheck/mce.c
index 3d6b5269fb2e,e4e6646cac46..0f8f21c8284a
--- a/arch/x86/kernel/cpu/mcheck/mce.c
+++ b/arch/x86/kernel/cpu/mcheck/mce.c
@@@ -52,11 -52,11 +52,11 @@@
  
  static DEFINE_MUTEX(mce_chrdev_read_mutex);
  
- #define rcu_dereference_check_mce(p) \
+ #define mce_log_get_idx_check(p) \
  ({ \
 -	rcu_lockdep_assert(rcu_read_lock_sched_held() || \
 -			   lockdep_is_held(&mce_chrdev_read_mutex), \
 -			   "suspicious mce_log_get_idx_check() usage"); \
 +	RCU_LOCKDEP_WARN(!rcu_read_lock_sched_held() && \
 +			 !lockdep_is_held(&mce_chrdev_read_mutex), \
- 			 "suspicious rcu_dereference_check_mce() usage"); \
++			 "suspicious mce_log_get_idx_check() usage"); \
  	smp_load_acquire(&(p)); \
  })